File: Flutter Dart Node Express - Orphanslife/lib/location.dart
import 'dart:convert';

List<Location> modelUserFromJson(String str) =>
    List<Location>.from(json.decode(str).map((x) => Location.fromJson(x)));

class Location {
  String _id, _pincode, _area, _city, _district, _state;

  Location.all(this._id,this._pincode,this._area,this._city,this._district,this._state);

  factory Location.fromJson(Map<String, dynamic> json) => Location.all(
      json["id"].toString(),
      json["pincode"].toString(),
      json["area"].toString(),
      json["city"].toString(),
      json["district"].toString(),
      json["state"].toString()
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "pincode": pincode,
    "area": area,
    "city" : city,
    "district" : district,
    "state" : state
  };

  get state => _state;

  set state(value) {
    _state = value;
  }

  get district => _district;

  set district(value) {
    _district = value;
  }

  get city => _city;

  set city(value) {
    _city = value;
  }

  get area => _area;

  set area(value) {
    _area = value;
  }

  get pincode => _pincode;

  set pincode(value) {
    _pincode = value;
  }

  String get id => _id;

  set id(String value) {
    _id = value;
  }

  @override
  String toString() {
    return 'Location{_id: $_id, _pincode: $_pincode, _area: $_area, _city: $_city, _district: $_district, _state: $_state}';
  }
}

File: MongoDB Express - API - Orphanslife/Services/Service_Extras/location.js
const express = require('express')
const utils = require('../../Functions/utils')
const auth = require('../../API_Auth/auth')
const { admin, editor, viewer } = require("../../API_Auth/roles.js");
const router = express.Router()
const Location = require('../../Data/model/Location.js')

router.get('/locations', [auth], async (req, res) => {
    try {
      const data = await Location.find();
      res.status(200).json(utils.createSuccessResult(data))
    }
    catch (error) {
        res.status(400).json(utils.createErrorResult(error))
    }
})

router.post('/findByIdLocation/:id', [auth], (request, response) => {
    const id = request.params.id
    console.log(id);
    const statement = `select * from location where id=${request.params.id};`
    db.pool.query(statement, [id], (error, result) => {
        response.send(utils.createResult(error, result))
    })
})

router.post('/findByPincodeLocation/:pincode', [auth], (request, response) => {
    const pincode = request.params.pincode
    console.log(pincode);
    const statement = `select * from location where pincode=${request.params.pincode};`
    db.pool.query(statement, [pincode], (error, result) => {
        response.send(utils.createResult(error, result))
    })
})

router.post('/newLocation/', [auth], (request, response) => {
    const {id,pincode,area,city,district,state} = request.body
    console.log(request.body);
    const statement = `insert into location values(?,?,?,?,?,?)`
    db.pool.query(statement, [id,pincode,area,city,district,state], (error, result) => {
        response.send(utils.createResult(error, result))
    })
})

module.exports = router